### PR TITLE
Add lots of structure to the installation guide

### PR DESCRIPTION
**Pull request by mattias-p**
The main features are:

* A dedicated "Dependencies" section
* A dedicated "Configuration" section
* A dedicated "Startup" section
* A dedicated "Post-installation sanity check" section

Having explicit sections for the above allows better integration with the coming
system testing documentation.

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -21,31 +21,10 @@ This document ends with these appendices:
 This installation guide assumes that the following softwares are already installed on the target system:
 
 * one of CentOS, Debian, FreeBSD or Ubuntu
-* one of MySQL or PostgreSQL
 * sudo
 * curl (only for post-installation sanity check)
```
Comment (matsduf) [inline docs/installation.md]: The databases have to be introduced before the dependency section. E.g.:

"This installation guide assumes that one of the possible database backends has been selected:
* Mysql
* Postgresql
* Sqlite"

File: docs/installation.md [hunk 2]
```diff
@@ -63,7 +42,11 @@ sudo cpan -i Config::IniFiles Daemon::Control JSON::RPC::Dispatch Parallel::Fork
 #### Installing dependencies for MySQL on CentOS
```
Comment (matsduf) [inline docs/installation.md]: Shouldn't it rather be "Installing Mysql and its dependencies on Centos"?

File: docs/installation.md (hunk 1 above)
Comment (mattias-p) [inline docs/installation.md]: This has been taken care of in #232.

File: docs/installation.md (hunk 2 above)
Comment (mattias-p) [inline docs/installation.md]: This has been taken care of in #232.

**Pull request merged**